ThitipatC iP

ThitipatC opened the pull request:
### **HAssnt**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

HAssnt frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/126).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)



### If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        launch(args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ThitipatC committed:
Change the bot name to HASSTNT

ThitipatC committed:
Create chatbot that echoes until bye is typed in

ThitipatC committed:
Rename chat bot to HASSTNT and change the skeleton

ThitipatC committed:
HASSTNT now can echo conversation

ThitipatC committed:
the following changes are made:
1. Restructuring the classes such that Main.java is the entry point and HASSTNT.java now becomes the used-to-be ChatbotFunctionality.java
2. HASSTNT now can keep and show the list of given input

ThitipatC committed:
HASSTNT now can mark and unmark task

ThitipatC committed:
HASSTNT now can handle different tasks being deadline, event, and todo

ThitipatC committed:
A-TextUiTesting

ThitipatC committed:
Already handle the exception, so just add some comments to be able to commit

ThitipatC committed:
HASSTNT now can delete elem from the todolist

ThitipatC committed:
Update conversation structure as well as adding errors

ThitipatC committed:
Now HASSNT could save todolist info from last session and reload it into another session

ThitipatC committed:
Now HASSNT only add Event and Deadline tasks which only contains the right date format

ThitipatC committed:
Restructure to make it adhere to OOP principle more

ThitipatC committed:
Restructure to make it adhere to OOP principle more

ThitipatC committed:
Group classes to packages

ThitipatC committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ThitipatC committed:
Added Gradle to be used in JUnit testing

ThitipatC committed:
Unit tested

ThitipatC committed:
added Javadoc

ThitipatC committed:
added Javadoc

ThitipatC committed:
modify code to comply to coding standard

ThitipatC committed:
added find command

ThitipatC committed:
Adjust the format to comply with the specified rules

ThitipatC committed:
Add GUI and a command to clear the list

xiaorui-ui commented on src/main/java/Commands/AddToListCommand.java:
```diff
@@ -0,0 +1,34 @@
+     *
+     * @param t The task to be added to the list.
+     */
+    public AddToListCommand(Task t) {
```
Perhaps use a more detailed name here (e.g. = task)?

xiaorui-ui commented on src/main/java/Commands/MarkCommand.java:
```diff
@@ -0,0 +1,32 @@
+ * The MarkCommand class represents a command to mark a task as done.
+ */
+public class MarkCommand extends Command {
+    private final int index;
```
Perhaps you could explain what this variable is for via Javadoc?

xiaorui-ui commented on src/main/java/Commands/ShowListCommand.java:
```diff
@@ -0,0 +1,26 @@
+        String s = taskList.showLists();
+        if (s.isEmpty()) {
+            return ui.showMessages("NO TASKS IN TO DO LIST 🎉");
+        } else {
```
Perhaps you wouldn't need the else statement? I feel like it makes the code cleaner

xiaorui-ui commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,51 @@
+/**
+ * The Task class represents a generic task.
+ */
+public class Task {
```
Have you considered making the class abstract? From the samples given in later levels it seems like all tasks need a type.

xiaorui-ui commented on src/main/java/UiRelated/Parser.java:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    private static Command commandWithNoIndex(String command) {
+        switch (command) {
```
Good!

xiaorui-ui commented on src/test/java/App/TestParser.java:
```diff
@@ -0,0 +1,82 @@
+/**
+ * Contains JUnit tests for the Parser class.
+ */
+public class TestParser {
```
The guide seems to suggest naming it ParserTest instead? I'm not too sure about this either

xiaorui-ui reviewed (commented):
LGTM with respect to coding style! Just a few nits to fix.

rizkidelta commented on src/main/java/App/Main.java:
```diff
@@ -0,0 +1,39 @@
+package App;
```
packages should be in lowercase

rizkidelta commented on src/main/java/UiRelated/Ui.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Displays the welcome message.
+     */
+    public void displayWelcomeMessage() {
```
Method names like displayWelcomeMessage, displayByeMessage, displayErrorMessage, and displayCommand are appropriately named, clearly describing their functionality and adhering to camelCase convention

rizkidelta commented on src/main/java/UiRelated/Ui.java:
```diff
@@ -0,0 +1,92 @@
+                                                + "-- our date format "
+                                                + "starts with month "
+                                                + "instead of day (i.e., mm-dd) --";
+    private static final String GOODBYE_MESSAGE = "HASSNT:\n" + "Bye. Hope to see you again soon!";
```
Constants like AVAILABLE_COMMAND, AVAILABLE_LIST_OPERATION, WELCOME_MESSAGE, and GOODBYE_MESSAGE are in all uppercase with underscores, following Java's naming convention for constants.

rizkidelta commented on src/main/java/Tasks/DeadLineTask.java:
```diff
@@ -0,0 +1,58 @@
+    public String logString() {
+        return "D" + super.logString() + "|" + deadLineLog;
+    }
+}
```
The implementation of the logString method in your DeadLineTask class is a commendable choice. It's a unique feature that greatly aids in debugging and monitoring, showing a thoughtful approach to differentiating between user-facing and log-specific representations. This enhances the maintainability and clarity of your code. Excellent work on this aspect!

rizkidelta commented on src/main/java/Commands/UnMarkCommand.java:
```diff
@@ -0,0 +1,32 @@
+     * @param ui       The Ui object for displaying messages.
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui) {
```
Good use of verb to describe action of method, this type of naming is seen all throughout the code

rizkidelta reviewed (commented):
Overall, your code demonstrates a strong adherence to Java best practices, particularly in naming conventions and method functionality. The thoughtful implementation of features like the logString method in DeadLineTask and consistent use of descriptive, verb-oriented method names like execute in UnMarkCommand showcase a clear and maintainable code structure. Your attention to detail in naming constants and ensuring package names follow standard conventions further emphasizes your commitment to coding standards. Excellent work overall!

ThitipatC committed:
Parser class: store a standalone boolean into a variable

direct use of method String:indexOf to check for valid pattern is harder to read and follow

Crate new boolean variables to store the boolean val of checking indexOf equals -1 into those

Using a properly named variable clearly illustrate the intention of if else clause

Parser:addCommand : divide part of the method into two helper command and incorporate those back to the addCommand method

those pats of the addCommand method are longer than 30 lines making the addCommand too long and unreadable

Create two helper function to handle the addtoCommand
for deadline and event tasks

Incorporating more helper functions make the addCommand more readable

ThitipatC committed:
add assertion into Parser:parseInput to ensure that empty string is handled

ThitipatC committed:
Merge pull request #1 from ThitipatC/branch-A-Assertions

add assertion into Parser:parseInput to ensure that empty string is h…

ThitipatC committed:
Merge branch 'branch-A-CodeQuality'

ThitipatC committed:
Add Detect Duplicate functionality to the TaskList:addToList

ThitipatC committed:
Add Detect Duplicate functionality to the TaskList:addToList

ThitipatC committed:
Add Detect Duplicate functionality to the TaskList:addToList

ThitipatC committed:
Update README.md

ThitipatC committed:
Add UI picture and update readme

ThitipatC committed:
Add UI picture and update readme

ThitipatC committed:
Update README.md

ThitipatC committed:
Update README.md

ThitipatC committed:
Update README.md

ThitipatC committed:
Update README.md

ThitipatC committed:
Update README.md

ThitipatC committed:
Update README.md

ThitipatC committed:
Update README.md
